src/libs/SearchUIUtils.ts

```diff
@@ -1704,6 +1715,12 @@ function getReportSections(
                 }
             }
 
+            const actionFromQuery = queryJSON?.flatFilters?.find((filter) => filter.key === CONST.SEARCH.SYNTAX_FILTER_KEYS.ACTION)?.filters?.at(0)?.value;
+
+            if (actionFromQuery && isValidActionFilter(actionFromQuery)) {
+                shouldShow = shouldShow && actionFilterMapping[actionFromQuery](reportItem);
+            }
```

Comment:
I think we should put this all inside the `if` above (`if (queryJSON && !isActionLoading)`)